go-controller/pkg/node/management-port_linux.go

```diff
@@ -23,59 +22,34 @@ const (
 // that lets the node access its pods via their private IP address. This is used
 // for health checking and other management tasks.
 func createPlatformManagementPort(interfaceName, interfaceIP, routerIP, routerMAC string) error {
-	// TODO - Migrate to using netlink.
-
-	// Up the interface.
-	_, _, err := util.RunIP("link", "set", interfaceName, "up")
+	link, err := util.GetLinkWithStateUp(interfaceName)
 	if err != nil {
 		return err
 	}
-
-	// The interface may already exist, in which case delete the routes and IP.
-	_, _, err = util.RunIP("addr", "flush", "dev", interfaceName)
-	if err != nil {
-		return err
-	}
-
-	// Assign IP address to the internal interface.
-	_, _, err = util.RunIP("addr", "add", interfaceIP, "dev", interfaceName)
+	// Flush any existing IP addresses and assign the new IP
+	err = util.LinkAddrAdd(link, interfaceIP)
 	if err != nil {
 		return err
 	}
 
+	// flush any existing routes and add new route for the cluster subnet
+	var clusterSubnets []string
 	for _, subnet := range config.Default.ClusterSubnets {
-		// Flush the route for the entire subnet (in case it was added before).
-		_, _, err = util.RunIP("route", "flush", subnet.CIDR.String())
-		if err != nil {
-			return err
-		}
-
-		// Create a route for the entire subnet.
-		_, stderr, err := util.RunIP("route", "add", subnet.CIDR.String(), "via", routerIP)
-		if err != nil {
-			if strings.HasPrefix(stderr, "RTNETLINK answers: File exists") {
-				klog.V(5).Infof("Ignoring error %s from 'route add %s via %s' - already added via IPv6 RA?",
-					strings.TrimSpace(stderr), subnet.CIDR.String(), routerIP)
-			} else {
-				return fmt.Errorf("Failed to add route for %s via %s: %s", subnet.CIDR.String(), routerIP, err)
-			}
-		}
+		clusterSubnets = append(clusterSubnets, subnet.CIDR.String())
 	}
-
-	// Flush the route for the services subnet (in case it was added before).
-	_, _, err = util.RunIP("route", "flush", config.Kubernetes.ServiceCIDR)
+	err = util.LinkRouteAdd(link, routerIP, clusterSubnets)
 	if err != nil {
 		return err
 	}
 
-	// Create a route for the services subnet.
-	_, stderr, err := util.RunIP("route", "add", config.Kubernetes.ServiceCIDR, "via", routerIP)
+	// flush any existing routes and add new route for the service subnet
+	err = util.LinkRouteAdd(link, routerIP, []string{config.Kubernetes.ServiceCIDR})
 	if err != nil {
-		if strings.HasPrefix(stderr, "RTNETLINK answers: File exists") {
+		if strings.Contains(err.Error(), "file exists") {
```

Comment:
@girishmg we should be checking the error with a type assertion here. But netlink helps becuase of the NLM_F_* flags so see below. But TLDR I think we can just remove this error check.

Reply:
ok